zlibWrapper/zstd_zlibwrapper.c

```diff
@@ -181,74 +246,92 @@ ZEXTERN int ZEXPORT z_deflate OF((z_streamp strm, int flush))
 {
     ZWRAP_CCtx* zwc;
 
-    if (!g_useZSTD) {
-        int res = deflate(strm, flush);
-        LOG_WRAPPER("- avail_in=%d total_in=%d total_out=%d\n", (int)strm->avail_in, (int)strm->total_in, (int)strm->total_out);
+    if (!g_ZWRAP_useZSTDcompression) {
+        int res;
+        LOG_WRAPPERC("- deflate1 flush=%d avail_in=%d avail_out=%d total_in=%d total_out=%d\n", (int)flush, (int)strm->avail_in, (int)strm->avail_out, (int)strm->total_in, (int)strm->total_out);
+        res = deflate(strm, flush);
         return res;
     }
 
     zwc = (ZWRAP_CCtx*) strm->state;
+    if (zwc == NULL) { LOG_WRAPPERC("zwc == NULL\n"); return Z_STREAM_ERROR; }
+
+    if (zwc->zbc == NULL) {
+        int res = ZWRAP_initializeCStream(zwc, (flush == Z_FINISH) ? strm->avail_in : 0);
```

Comment:
My understanding here is that `ZWRAP_initializeCStream()` is invoked with the size of first source _if available_ `(flush == Z_FINISH)`.

So I guess that's where start the discussion : 
what happens if first source is very small, resulting in very small compression parameters, but later sources using the same context are much larger, resulting in poor compression ratios for these later sources ?

I guess the impact will be more clearly visible if there are some tests to show it.